builtin/logical/transit/path_sign_verify.go

```diff
@@ -225,7 +236,38 @@ func (b *backend) pathVerifyWrite(
 		return logical.ErrorResponse("encryption key not found"), logical.ErrInvalidRequest
 	}
 
-	valid, err := p.VerifySignature(hashedInput, sig)
+	if !p.Type.SigningSupported() {
+		return logical.ErrorResponse(fmt.Sprintf("key type %v does not support verification", p.Type)), logical.ErrInvalidRequest
+	}
+
+	contextRaw := d.Get("context").(string)
```

Comment:
This code block looks the same as the one above, can it be turned into a `func` call?